feat(lgr): create lgr backend

GauChoob opened the pull request:
Here is a backend for storing and retrieving lgr files.

Let me know if this is alright or if there are changes that you suggest!

~sunl

GauChoob commented:
I'm more or less done
Will need to sync the database for the following models: LGR, LGRTags, Tag, LGRComment, Notification
Will also need to add some tags for lgr (e.g. "Alt Palette" (Hidden=yes), "Default", "Theme")

sunehs commented on src/api/lgr.js:
```diff
@@ -0,0 +1,370 @@
+});
+
+// pipes a .lgr or preview.img file
+const getLGRFile = async (req, res, next, key) => {
```
any reason to pipe it through the backend rather than downloading from the url directly? (other than the local option)

GauChoob replied:
The other question is can you cache the lgrs files when downloaded directly from S3? They are put in an lgr/ subfolder

sunehs replied:
not sure what you mean with direct link to database

and yes we can change cache settings for s3 files, either manually after upload or lilkely with some parameters when uploading

sunehs replied:
Anyway, keeping the endpoint is fine

GauChoob commented:
Added a small feature where mods can change ownership of lgrs. I also added cache controls to lgr files (can't really test it though since I don't have access to S3)
